.github/workflows/update-version.yaml

```diff
@@ -0,0 +1,47 @@
+name: Update Version
+on:
+  workflow_dispatch:
+    inputs:
+      current_beta:
+        description: 'Current beta version, e.g. v0.111.0 or 0.111.0'
+        required: false
+        default: ''
+      current_stable:
+        description: 'Current stable version, e.g. v1.16.0 or 1.16.0'
+        required: false
+        default: ''
+      next_beta:
+        description: 'Next beta version (optional)'
+        required: false
+        default: ''
+      next_stable:
+        description: 'Next stable version (optional)'
+        required: false
+        default: ''
+
+jobs:
+  update-version:
+    runs-on: ubuntu-latest
+    env:
+      GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - name: Validate inputs
+        run: |
+          if [ -z "${{ github.event.inputs.current_beta }}" ] && [ -z "${{ github.event.inputs.current_stable }}" ]; then
+            echo "Error: At least one of 'current_beta' or 'current_stable' must be provided."
+            exit 1
+          fi
+
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Run bump-versions.sh
+        run: |
+          chmod +x .github/workflows/scripts/bump-versions.sh
+          .github/workflows/scripts/bump-versions.sh \
+            --current_beta "${{ github.event.inputs.current_beta }}" \
+            --current_stable "${{ github.event.inputs.current_stable }}" \
+            --next_beta "${{ github.event.inputs.next_beta }}" \
+            --next_stable "${{ github.event.inputs.next_stable }}"
```

Comment:
As mentioned in https://github.com/open-telemetry/opentelemetry-collector-releases/pull/684#pullrequestreview-2345254646, there are three distinct versions used in the builder configurations. Two of those are equal _most of the time_. This script assumes that core and contrib versions are always the same, which means it will do the wrong thing in case there's a mismatch between core and contrib versions.

I'm reluctant to have an automation in place that works correctly _most of the time_, as opposed to _always_. Releasing software should be as frictionless as possible, given that it is sometimes done under pressure (e.g. for hot fix releases).

I propose to update the script to accept three versions, and perform the replacements separately for core and contrib components. It is possible to easily tell those apart as the module paths are different.

Reply:
apologies, I must not have read your comment thoroughly enough. I agree with your proposed change and will push an update to the PR branch shortly.